src/layout/layout.ts

```diff
@@ -148,11 +148,13 @@ class Layout implements IComponent {
   private onDomDiscoverComplete() {
     this.domDiscoverComplete = true;
 
-    let allMutationRecords: MutationRecord[] = [];
-    for (let i = 0; i < this.domPreDiscoverMutations.length; i++) {
-      allMutationRecords = allMutationRecords.concat(this.domPreDiscoverMutations[i].mutations);
+    if (this.domPreDiscoverMutations.length > 0) {
+      let allMutationRecords: MutationRecord[] = [];
+      for (let i = 0; i < this.domPreDiscoverMutations.length; i++) {
```

Comment:
would concat.apply(this, domPreDiscovreMutations) be faster and simpler? concat() can concatenate multiple arrays.